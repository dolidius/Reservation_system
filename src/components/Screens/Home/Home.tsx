import React, { useEffect } from 'react';

import { connect } from 'react-redux';

import { getSeats } from '../../../redux/actions/seatsActions';

import { Form, InputNumber, Button, Checkbox } from 'antd';

import { RouteComponentProps } from 'react-router-dom';

import { ViewportContainer } from '../../../styles/Layout/Container.style';

interface IFormValues {
    tickets_number: number;
    next_to_each_other: boolean;
}

interface IProps {
    history: RouteComponentProps['history'];
    getSeats: () => void;
}

const Home: React.FC<IProps>  = ({ history, getSeats }) => {

    const [form] = Form.useForm();

    const onFinish = (values: IFormValues) => {
        getSeats();
        history.push("/rezerwacja");
    }

    return (
        <ViewportContainer centered>
            <Form form={form} name="tickets" onFinish={onFinish}>
                <Form.Item name="tickets_number" rules={[{ required: true }]}>
                    <InputNumber />
                </Form.Item>
                <Form.Item name="next_to_each_other" valuePropName="checked">
                    <Checkbox>
                        Czy miejsca mają być obok siebie?
                    </Checkbox>
                </Form.Item>
                <Form.Item>
                    <Button type="primary" htmlType="submit">
                        Submit
                    </Button>
                </Form.Item>
            </Form>
        </ViewportContainer>
    )
}

const mapStateToProps = () => ({});

export default connect((mapStateToProps), { getSeats })(Home);