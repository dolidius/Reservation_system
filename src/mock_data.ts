export const seats =  [
    {
      "id": "s02",
      "cords": {
        "x": 0,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s03",
      "cords": {
        "x": 0,
        "y": 3
      },
      "reserved": false
    },
    {
      "id": "s04",
      "cords": {
        "x": 0,
        "y": 4
      },
      "reserved": false
    },
    {
      "id": "s06",
      "cords": {
        "x": 0,
        "y": 6
      },
      "reserved": false
    },
    {
      "id": "s07",
      "cords": {
        "x": 0,
        "y": 7
      },
      "reserved": false
    },
    {
      "id": "s08",
      "cords": {
        "x": 0,
        "y": 8
      },
      "reserved": true
    },
    {
      "id": "s09",
      "cords": {
        "x": 0,
        "y": 9
      },
      "reserved": true
    },
    {
      "id": "s011",
      "cords": {
        "x": 0,
        "y": 11
      },
      "reserved": false
    },
    {
      "id": "s012",
      "cords": {
        "x": 0,
        "y": 12
      },
      "reserved": false
    },
    {
      "id": "s013",
      "cords": {
        "x": 0,
        "y": 13
      },
      "reserved": false
    },
    {
      "id": "s014",
      "cords": {
        "x": 0,
        "y": 14
      },
      "reserved": false
    },
    {
      "id": "s12",
      "cords": {
        "x": 1,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s13",
      "cords": {
        "x": 1,
        "y": 3
      },
      "reserved": false
    },
];