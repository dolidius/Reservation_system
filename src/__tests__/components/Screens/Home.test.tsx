import { Provider } from "react-redux";
import Home from "../../../components/Screens/Home/Home";

import { createMemoryHistory } from 'history';

import * as renderer from "react-test-renderer";

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { seats } from "../../../mock_data";

const middlewares = [thunk];

const mockStore = configureStore(middlewares);

const history = createMemoryHistory();

describe("<Home />", () => {
    it("should return booking form correctly", () => {
        let store = mockStore({
            seats: {
                loading: false,
                seats,
                seatsToChoose: 0,
                nextToEachOther: false,
                error: false,
                errorMessage: "",
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <Home history={history} />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("should render booking form with error when its on store", () => {

        let store = mockStore({
            seats: {
                loading: false,
                seats,
                seatsToChoose: 0,
                nextToEachOther: false,
                error: true,
                errorMessage: "Nie można zarezerwować 0 biletów",
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <Home history={history} />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();

    });

});
