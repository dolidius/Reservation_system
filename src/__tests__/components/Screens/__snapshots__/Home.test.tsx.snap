// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render booking form with error when its on store 1`] = `
<div
  className="sc-gtsrHT cJzSkO"
>
  <form
    className="sc-dlnjwi hANKYa"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="filled-required"
        id="filled-required-label"
      >
        Liczba Miejsc
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="filled-required"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="0"
          required={true}
          type="number"
          value="0"
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-7"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorError"
      >
        Nie można zarezerwować 0 biletów
      </p>
    </div>
    <div
      className="MuiBox-root MuiBox-root-8"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input-5"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Czy miejsca mają być obok siebie?
          </p>
        </span>
      </label>
    </div>
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Wybierz miejsca
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`<Home /> should return booking form correctly 1`] = `
<div
  className="sc-gtsrHT cJzSkO"
>
  <form
    className="sc-dlnjwi hANKYa"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
        data-shrink={true}
        htmlFor="filled-required"
        id="filled-required-label"
      >
        Liczba Miejsc
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="filled-required"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="0"
          required={true}
          type="number"
          value="0"
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input-5"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Czy miejsca mają być obok siebie?
          </p>
        </span>
      </label>
    </div>
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Wybierz miejsca
        </span>
      </button>
    </div>
  </form>
</div>
`;
